### PR TITLE
updating event page

### PR DESCRIPTION
**Pull request by odicho**
## Description
- Refactor how to get challengeId's: Now using trackChallenge mapping
- Added basic skeleton code for events page to get all challenges so far and render the rest of days with some sample jsx

### CODE REVIEW

File: apps/aot/src/app/leaderboard/getAotChallengeIds.ts [hunk 1]
```diff
@@ -4,32 +4,32 @@ import { getCurrentAdventDay } from '~/utils/time-utils';
...
   const currentAdventDay = getCurrentAdventDay();
-  const year = new Date().getUTCFullYear().toString();
-  const day = String(currentAdventDay);
-  const endSlug = getAotSlug({ year, day });
+  // TODO: Put leaderboard under event/[year]/leaderboard
+  // pass in year from slug
+  const year = new Date().getUTCFullYear();
+
+  // Get the starting id of this year (first aot was 2023)
+  const startId = (year - 2023) * 25 + 1;
```
Comment (bautistaaa) [inline apps/aot/src/app/leaderboard/getAotChallengeIds.ts:12]: not following how this works 🤔

File: apps/aot/src/server/api/routers/event.ts [hunk 2]
```diff
@@ -80,33 +81,31 @@ export const eventRouter = createTRPCRouter({
...
-          slug: `advent-of-typescript-${input.year}`,
-        },
-        select: {
-          trackChallenges: {
-            orderBy: {
-              orderId: 'asc',
-            },
+      const currentAdventDay = getCurrentAdventDay();
+      // Get the starting id of this year (first aot was 2023)
+      const startId = (Number(input.year) - 2023) * 25 + 1;
```
Comment (bautistaaa) [inline apps/aot/src/server/api/routers/event.ts:86]: if year was 2025 the startId would be 51? based on what?

File: apps/aot/src/app/leaderboard/getAotChallengeIds.ts (hunk 1 above)
Comment (odicho) [reply apps/aot/src/app/leaderboard/getAotChallengeIds.ts:12]: The TrackChallenge table will have id's starting from 1 and auto incrementing for each challenge released each day of advent. So first year of advent of typehero was in 2023.

E.g. 2023's id's will always be 1-25, 2024 will be 26-50, etc

File: apps/aot/src/utils/time-utils.ts [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
-  return Date.now() >= aoStartTime.getTime();
+
+  return today >= aoStartTime.getTime();
 }
 
 export function getCurrentAdventDay() {
   const MS_PER_DAY = 24 * 60 * 60 * 1000;
 
-  const today = Date.now();
+  const today = new Date().getTime();
```
Comment (bautistaaa) [inline apps/aot/src/utils/time-utils.ts:11]: does this need to be on EST as well?

File: apps/aot/src/app/leaderboard/getAotChallengeIds.ts (hunk 1 above)
Comment (bautistaaa) [reply apps/aot/src/app/leaderboard/getAotChallengeIds.ts:12]: that's not true though. we have other tracks outside of advent of typescript that could intertwine with the ranges

File: apps/aot/src/utils/time-utils.ts (hunk 3 above)
Comment (odicho) [reply apps/aot/src/utils/time-utils.ts:11]: All these dates are in UTC! 

Also, I am going to change these a bit soon when I migrate everything to be under events/[year]/

getCurrentAdventDay and hasAdventStarted will use the year from the slug and I will use this syntax to make everything clear: example: `new Date(Date.UTC(2024, 11, 1, 5, 0, 0));`

File: apps/aot/src/app/leaderboard/getAotChallengeIds.ts (hunk 1 above)
Comment (odicho) [reply apps/aot/src/app/leaderboard/getAotChallengeIds.ts:12]: Oh lol 😆 😭

**Review by bautistaaa — APPROVED** [review]

**Pull request merged**